src/uwtools/api/config.py

```diff
@@ -84,8 +84,7 @@ def realize(
     input_format: Optional[str] = None,
     output_file: OptionalPath = None,
     output_format: Optional[str] = None,
-    values: Union[dict, _Config, OptionalPath] = None,
-    values_format: Optional[str] = None,
+    supplemental_configs: Optional[List[Union[dict, _Config, DefinitePath]]] = None,
     values_needed: bool = False,
     dry_run: bool = False,
 ) -> bool:
```

Comment:
Regarding the order of arguments, I'm unsure of two things: 
1) because `supplemental_configs` takes in positional arguments, should it be placed last in this call (and all similar calls)? 
2) what is the reason behind `values_needed` being listed above `dry_run`?

Reply:
1. If you mean "positional arguments" with respect to arguments given on the command line, I don't think that concept is still meaningful at this point in the code. After `argparse` parses the command-line arguments in `cli.py`, we get a `Namespace` object containing the parsed arguments, and I don't think we need to treat any of them differently than the others after that, except where it's helpful for us, or makes for a sensible function/method signature. Python only demands that mandatory parameters precede optional ones and, since all the parameters here are optional, the order is up to us. I think best practice is for the most important / meaningful parameters to come first, with less important (and, when there's a distinction, optional) parameters coming later, with those least likely to be of interest to most users coming last. So, here, I guess I see the inputs and outputs as being the "main parameters" (in fact, they're kind of mandatory, in that `stdin` and `stdout` will be used if they're not specified -- we're definitely going to do input and output whether we get values for these arguments or not), with the supplemental configs following in importance, and then the useful but one-off `values_needed` coming next, and then the fairly-fringe `dry_run` coming last.
2. About `dry_run`, another consideration is that many of our functions/methods have a `dry_run` parameter, and we've so far been pretty consistent about having it appear late-to-last in the signature. It's almost an afterthought, and doesn't have much to do with the purpose of the function. Of course it's there, but it's not in your face because it's not very important. If we let it appear in different positions for different functions, it would IMO get in the way of looking at function signatures and understanding what they do. I hope that makes some sense and is vaguely convincing. Please speak up if you disagree!

Reply:
Interestingly the two are kind of tied together - `dry_run` was stuck at the end where it would usually sit with `quiet` and `verbose` as kind of operands. Though, shouldn't operands go before options? That would put `dry_run` and `values_needed` first and `supplemental_configs` last